dora/core/common/src/main/java/alluxio/file/NettyBufTargetBuffer.java

```diff
@@ -37,7 +37,9 @@ public NettyBufTargetBuffer(ByteBuf target) {
 
   @Override
   public byte[] byteArray() {
-    throw new UnsupportedOperationException();
+    byte[] bytes = new byte[mTarget.readableBytes()];
+    mTarget.readBytes(bytes, 0, bytes.length);
+    return bytes;
```

Comment:
this is copying the data from the buffer into a byte array. the interface however is intended to allow zero-copy access of the contained data.
you can call `ByteBuf.array()` to expose the underlying buffer, if any, or the method would throw `UnsupportedOperationException` if the bytebuf is not backed by an byte array.